extensions/AzureCosmosDBMongoDB/AzureCosmosDBMongoDB/AzureCosmosDBMongoDBMemory.cs

```diff
@@ -0,0 +1,309 @@
+    {
+        AzureCosmosDBMongoDBMemoryRecord localRecord = AzureCosmosDBMongoDBMemoryRecord.FromMemoryRecord(record);
+        var filter = Builders<AzureCosmosDBMongoDBMemoryRecord>.Filter.Eq(r => r.Id, localRecord.Id);
+
+        var replaceOptions = new ReplaceOptions() { IsUpsert = true };
+
+        await this._cosmosMongoCollection.ReplaceOneAsync(filter, localRecord, replaceOptions, cancellationToken).ConfigureAwait(false);
+
+        return record.Id;
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<(MemoryRecord, double)> GetSimilarListAsync(
+        string index,
+        string text,
+        ICollection<MemoryFilter>? filters = null,
+        double minRelevance = 0,
+        int limit = 1,
+        bool withEmbeddings = false,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        if (limit <= 0) { limit = int.MaxValue; }
+
+        Embedding embedding = await this._embeddingGenerator.GenerateEmbeddingAsync(text, cancellationToken).ConfigureAwait(false);
+
+        BsonDocument[]? pipeline = null;
+        switch (this._config.Kind)
+        {
+            case AzureCosmosDBVectorSearchType.VectorIVF:
+                pipeline = this.GetVectorIVFSearchPipeline(embedding, limit);
+                break;
+            case AzureCosmosDBVectorSearchType.VectorHNSW:
+                pipeline = this.GetVectorHNSWSearchPipeline(embedding, limit);
+                break;
+        }
+
+        using var cursor = await this._cosmosMongoCollection.AggregateAsync<BsonDocument>(pipeline, cancellationToken: cancellationToken).ConfigureAwait(false);
+        while (await cursor.MoveNextAsync(cancellationToken).ConfigureAwait(false))
+        {
+            foreach (var doc in cursor.Current)
+            {
+                // Access the similarityScore from the BSON document
+                var similarityScore = doc.GetValue("similarityScore").AsDouble;
+                if (similarityScore < minRelevance) { continue; }
+
+                MemoryRecord memoryRecord = AzureCosmosDBMongoDBMemoryRecord.ToMemoryRecord(doc, withEmbeddings);
+
+                yield return (memoryRecord, similarityScore);
+            }
+        }
+    }
+
+    /// <inheritdoc />
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+    public async IAsyncEnumerable<MemoryRecord> GetListAsync(
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+        string index,
+        ICollection<MemoryFilter>? filters = null,
+        int limit = 1,
+        bool withEmbeddings = false,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        // TODO: Add logic for search with filters, once it is released in Azure Cosmos DB for MongoDB.
+        // For now this method returns an empty list.
```

Comment:
does it mean the connector is still underdevelopment?

Reply:
The filters are in public preview now, I'll add it in the next iteration in a couple of weeks. Does this block the PR right now?

Reply:
Without filtering the connector would not be useful, because it would prevent from using security filters